### ISSUE TITLE
<abbr> tag component

### ISSUE THREAD

**Issue by DanGuyMOJ** [alphagov/govuk-design-system#1289]
In an assessment, it was recommended that our service implement "Acronym Markdown" / <abbr> tags for accessibility purposes. These are currently in use on the following page:

https://www.gov.uk/power-of-attorney

<img width="1280" alt="acronym-markdown" src="https://user-images.githubusercontent.com/65242377/84513788-e17db900-acc1-11ea-9a44-f51f36abbe58.png">

However, it is unclear that acronyms can be hovered and there is no guidance on how to use abbreviation tags in the design system. If this is a component worth adding to the design system then perhaps there could be some sort of visual cue that they can be hovered along with guidance on how to use the component.

**Comment by edwardhorsford** [comment]
On www.gov.uk, they _used_ to get different styling - they had a dotted underline. I wonder if we could pull up the commit history to find out why they were changed.

**Comment by timpaul** [comment]
This requires further investigation, to understand what accessibility issues there are with the standard browser vendor implementations of the <abbr> tag.

**Comment by StephenGill** [comment]
> I think the stance we've evolved is to not rely on <abbr> tags with titles at all, and to instead define the meanings of acronyms in their first instance in prose.

This makes sense for Iong-form content, but I think the considerations are little different for form-type services.

The 'one thing per page' principle means that it's common for these services to have a lot of screens with very few words on each.

At the moment, you have to choose between using the abbreviation without explanation - or adding significant amounts of repetitive microcopy. Plus often, when you're using a service knowing the full name of the thing isn't especially important in completing your goal. For example, if you tell me to "sign the copy of the LPA we send to you", that's probably no clearer than saying "sign the copy of the Lasting Power of Attorney we send to you". Because for most people, 'Lasting Power of Attorney' will still be an unfamiliar concept that needs explaining.

So if there's a way of solving the problems with zoom and verbosity, an <abbr> tooltip or something similar seems like a useful halfway house between using the abbreviation without explanation and spelling it out on every page.

**Comment by querkmachine** [comment]
> So if there's a way of solving the problems with zoom and verbosity, an tooltip or something similar seems like a useful halfway house

Resolving these on our own terms would likely require creating a new component that completely overrides how `<abbr>` works, but this would very likely introduce its own limitations, such as not allowing abbreviations inside links, labels or buttons, or 'breaking' how assistive technologies (particularly screen readers and voice control software) interact with the element.

And that's sort of the crux of the issue — we alone can't satisfactorily fix this. These are issues inhrent with how operating systems, user agents and assistive technologies handle the `<abbr>` element and native tooltips, and how they each interact with one another. Ideally, it would be the vendors that make this properly usable and properly accessible in the first place!

**Comment by davidc-gds** [comment]
I'll chime in to note that the [Understanding document for WCAG success criterion 'Abbreviations (AAA)'](https://www.w3.org/WAI/WCAG22/Understanding/abbreviations.html#sufficient) lists a few potential different techniques. 

It would be great to find some way to meet the 'Abbreviations' criterion, despite it being at the AAA level. It's one of the few WCAG 2.x criteria that addresses cognitive accessibility needs.

But more of note to this thread specifically, there's a ['sufficient' technique that involves using the `<abbr>` element](https://www.w3.org/WAI/WCAG22/Techniques/html/H28).

Some pros of `<abbr>`:
- can be used alongside many other techniques, such as spelling out the abbreviation the firs time it's used on a page
- programmatically determinable
- built into the HTML spec
- can be ingested by various technologies in whatever way they determine
- can be a good enhancement in places where abbreviations cannot be removed
- helps people on desktop computers

Some cons of `<abbr>` (most already mentioned by @querkmachine):
- relies on hover as the default interaction on most browsers
- pretty bad on most touchscreen interfaces
- doesn't play well with zoom
- screen readers can only really handle them in an 'all or nothing' way
- standard styling (dotted underline) is messy; clashes with links
- encourages or endorses use of abbreviations overall
